theatre/studio/src/panels/SequenceEditorPanel/layout/tree.ts

```diff
@@ -14,12 +14,25 @@ import {prism, val} from '@theatre/dataverse'
 import logger from '@theatre/shared/logger'
 import {titleBarHeight} from '@theatre/studio/panels/BasePanel/common'
 
-export type SequenceEditorTree_Row<Type> = {
-  type: Type
+/**
+ * Base "view model" for each row with common
+ * required information such as row heights & depth.
+ */
+export type SequenceEditorTree_Row<TypeName extends string> = {
+  /** type of this row, e.g. `"sheet"` or `"sheetObject"` */
+  type: TypeName
+  /** Height of just the row in pixels */
   nodeHeight: number
+  /** Height of the row + height with children in pixels */
   heightIncludingChildren: number
+  /** Visual indentation */
   depth: number
+  /**
+   * Distance in pixels from the top of this row to the row container's top
+   * This can be used to help figure out what's being box selected (marquee).
+   */
   top: number
+  /** Row number (e.g. for correctly styling even / odd alternating styles) */
   n: number
 }
```

Comment:
@vezwork, look right?

Reply:
looks great!

Reply:
nit: depth is not exactly visual indentation I don't think, but it is used to calculate visual indentation.

Reply:
I think `depth` indicates the level of nesting of a Sheet Object ([source](https://github.com/theatre-js/theatre/blob/aa1635bc3f7219e8a145a4c4570d97adbb53dbbe/theatre/studio/src/panels/SequenceEditorPanel/layout/tree.ts#L113)) in the tree. 

And yes, later it is used to calculate visual indentation (its value is used to set the `--depth` css variable ([source](https://github.com/theatre-js/theatre/blob/aa1635bc3f7219e8a145a4c4570d97adbb53dbbe/theatre/studio/src/panels/SequenceEditorPanel/DopeSheet/Left/AnyCompositeRow.tsx#L10))), which is later used to calculate the `padding-left` property of the rows ([source](https://github.com/theatre-js/theatre/blob/aa1635bc3f7219e8a145a4c4570d97adbb53dbbe/theatre/studio/src/panels/SequenceEditorPanel/DopeSheet/Left/AnyCompositeRow.tsx#L24)):

```css
  padding-left: calc(8px + var(--depth) * 20px);
```

Reply:
@fulopkovacs & @vezwork, depth is only used for visual indentation, but it is currently derived from child nestings. 

Key example is when implementing namespacing–the depth no longer means the same as depth within the tree; it indicates visual indentation.